### PR TITLE
fix(query-performance): Limit /api/event usage

### PR DESCRIPTION
**Pull request by macobo**
## Problem

Small teams using /api/event can cause us a _lot_ of clickhouse load. The api is not suited for large-scale exports, only for the live events table

## Changes

If using personal API key on cloud, raise a 400 error if filtering on over 3 days. This should stem the bleeding.

Also added instrumentation for clickhouse queries for separating requests from personal api keys from ones without. 

## How did you test this code?

Only using unit tests.

### CODE REVIEW

File: posthog/auth.py [hunk 1]
```diff
@@ -94,7 +94,12 @@ def authenticate(cls, request: Union[HttpRequest, Request]) -> Optional[Tuple[An
         assert personal_api_key_object.user is not None
 
         # :KLUDGE: CHMiddleware does not receive the correct user when authenticating by api key.
-        tag_queries(user_id=personal_api_key_object.user.pk, team_id=personal_api_key_object.user.current_team_id)
+        tag_queries(
+            user_id=personal_api_key_object.user.pk,
+            team_id=personal_api_key_object.user.current_team_id,
+            access_method="personal_api_key",
+        )
+        request.using_personal_api_key = True  # type: ignore
```
Comment (mariusandra) [inline posthog/auth.py:102]: I wonder if we should get these lines in first in a separate PR, and monitor for a bit. We spotted 5 users on `/api/event` yesterday, out of whom only the one troubled user would be affecter. However are there perhaps a few dozen users using `/api/project/x/event` that we don't know of, who'd also be affected?

File: posthog/models/event/query_event_list.py [hunk 2]
```diff
@@ -29,20 +30,32 @@ class EventsQueryResponse:
...
+
+    after_date = conditions.get("after")
+    before_date = conditions.get("before")
+    default_after = None if unbounded_date_from else now() - timedelta(days=1)
+    after = isoparse(after_date) if isinstance(after_date, str) else default_after
+    default_before = after + timedelta(days=1, seconds=5) if after is not None else now() + timedelta(seconds=5)
+    before = isoparse(before_date) if isinstance(before_date, str) else default_before
+
+    if validate_date_range and (after is None or before - after >= timedelta(days=3)):
+        raise ValidationError("Cannot query time range larger than 3 days using personal API key.")
```
Comment (lharries) [inline posthog/models/event/query_event_list.py:47]:
```suggestion
        raise ValidationError("Cannot query time range larger than 3 days using personal API key. If you are looking to export large amounts of data we recommend using the historical exports e.g. Amazon S3 export")
```

File: posthog/auth.py (hunk 1 above)
Comment (macobo) [reply posthog/auth.py:102]: https://github.com/PostHog/posthog/pull/13443

**Review by mariusandra — APPROVED** [review]
Code looks correct. If/when we're sure this won't affect many or any users who depend on this behaviour, feel free to merge. So here's a ✅ 

However, I keep thinking that considering how much time we've already spent on this, an API that makes a few incremental queries to get results up to the limit doesn't sound like that much more work. Basically, always first fetch the last 3 days, no matter if using the default, or a provided before/after range. The cleverest system could do this in a loop, fetching 3, 9, 27, 81, etc days and adding the results together until it gets `$limit`. Maybe... I'll be happy to do this in some followup as well, when it becomes a blocker for data exploration.

**Pull request closed without merging**